spec/lib/payment_processor/go_cardless/error_processing_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
+module PaymentProcessor
+  module GoCardless
+    describe ErrorProcessing do
+
+      let(:client) do
+        GoCardlessPro::Client.new(
+          access_token: Settings.gocardless.token,
+          environment:  Settings.gocardless.environment.to_sym
+        ) 
+      end
+
+      before :each do
+        allow(Rails.logger).to receive(:error)
+      end
+
+      subject{ described_class.new(@error).process }
+
+      describe 'with a GoCardless internal error', :pending do
+
+        before :each do
+          begin
+            client.redirect_flows.create(params: {})
+          rescue GoCardlessPro::GoCardlessError => e
+            @error = e
+          end
```

Comment:
this is essentially my problem - I don't know how to instantiate the GoCardless errors without getting the library to raise them for me. they do it in the lib by calling `fail error_class, response['body']` with the api response. If I use `GoCardlessError.new(params)` then the params don't make it to the accessor methods or stack trace. but if I try to use the library to raise the errors, I end up making http requests to GC and I don't want VCR in my unit spec.